ibis/backends/tests/test_string.py

```diff
@@ -1023,3 +1023,28 @@ def test_multiple_subs(con):
     expr = ibis.literal("foo").substitute(m)
     result = con.execute(expr)
     assert result == "FOO"
+
+
+@pytest.mark.notimpl(
+    [
+        "clickhouse",
+        "dask",
+        "datafusion",
+        "druid",
+        "impala",
+        "mssql",
+        "mysql",
+        "pandas",
+        "polars",
+        "sqlite",
+    ],
+    raises=com.OperationNotDefinedError,
+)
+@pytest.mark.parametrize(
+    "right", ["sitting", ibis.literal("sitting")], ids=["python", "ibis"]
```

Comment:
@cpcloud IDK if you want to be this careful, but there is a very similar Damerau-Levenshtein metric that counts transpositions as one edit, so maybe we want to verify that the backend is using the levenshtein, and not damerau-levenshtein

levenshten("iktton", "kitten") -> 3 
damerau_levenshten("iktton", "kitten") -> 2 (swap ik, then change e to o)

Reply:
@NickCrews Yeah! I think we can always add the Damerau-Levenshtein API later if desired.

Reply:
I realize I'm super late for this discussion, and I agree that the name `edit_dist` is vague, but perhaps we could use it with an optional parameter `type`? I'm thinking of:

`edit_distance(type='levenshtein') `

The possible options for type would be backend-dependent. 

This has at least two points in favor:

- Will allow for the inclusion of backends that do not define Levenshtein, for example (pandas with the built-in Ratcliff-Obershelp distance from [difflib](https://docs.python.org/3/library/difflib.html)) 
- For backends that support more than one type (like [DuckDB](https://duckdb.org/docs/sql/functions/char.html#text-similarity-functions)) allow the user to select the type without creating additional operations.